packages/core/src/command/ai-brief/shared.ts

```diff
@@ -24,7 +26,7 @@ export function generateAiBriefFhirResource(content: string | undefined): Binary
 
 export function getAiBriefContentFromBundle(bundle: Bundle): string | undefined {
   const binaryResourceEntry = bundle.entry?.find(
-    entry => entry.resource?.resourceType === "Binary"
+    entry => isBinary(entry.resource) && entry.resource?.meta?.source === AI_BRIEF_SOURCE
   );
 
   const resource = binaryResourceEntry?.resource;
```

Comment:
The is Binary is necessary below as well because the above find doesnt apply the casting

Reply:
We can use `.flatMap` to return an array of Binary and then filter by those that have the source we want.

But as I wrote the code to suggest it, I noticed we could do even better by writing a dedicated function to get the AiBriefResource (Binary + expected source), and then noticed we could have more than one, etc.... so I ended up with this recommendation:

```ts
export function getAiBriefContentFromBundle(bundle: Bundle): string | undefined {
  const aiBriefResource = getAiBriefResource(bundle);
  if (!aiBriefResource?.data) return undefined;
  return base64ToString(aiBriefResource.data);
}

function getAiBriefResource(bundle: Bundle): Binary | undefined {
  const { log } = out("getAiBriefContentFromBundle");
  const aiBriefResources =
    bundle.entry?.flatMap(entry => (isAiBriefResource(entry.resource) ? entry.resource : [])) ?? [];
  if (aiBriefResources.length > 1) {
    const msg = `Found more than one AI brief resource in the consolidated bundle`;
    const patient = getPatientFromBundle(bundle, false);
    log(`${msg} - ${aiBriefResources.length} AI Briefs, patient: ${patient?.id}`);
    capture.message(msg, { extra: { aiBriefResources, patient } });
  }
  return aiBriefResources[0];
}

function isAiBriefResource(resource: Resource | undefined): resource is Binary {
  return isBinary(resource) && resource.meta?.source === AI_BRIEF_SOURCE;
}

// TODO move to packages/core/src/external/fhir/patient/shared.ts
export function getPatientFromBundle(bundle: Bundle, warnOnMultiple = true): Patient | undefined {
  const patients = bundle.entry?.flatMap(entry =>
    isPatient(entry.resource) ? entry.resource : []
  );
  if (!patients?.length) return undefined;
  if (patients.length > 1 && warnOnMultiple) {
    const { log } = out("getPatientFromBundle");
    const msg = "Found multiple Patient resources in bundle";
    log(`${msg}, patientCount: ${patients.length}`);
    capture.message(msg, { extra: { patientIds: patients.map(p => p.id) } });
  }
  return patients[0];
}
```